extensions/librdkafka/PublishKafka.cpp

```diff
@@ -639,6 +647,17 @@ void PublishKafka::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
     PublishKafka::ReadCallback callback(max_flow_seg_size, kafkaKey, thisTopic->getTopic(), conn->getConnection(), flowFile,
                                         attributeNameRegex, messages, flow_file_index);
     session->read(flowFile, &callback);
+
+    bool dropEmptyFlowFiles;
+    if (!callback.called_ && context->getProperty(DropEmptyFlowFiles.getName(), dropEmptyFlowFiles) && !dropEmptyFlowFiles) {
+      // workaround: call callback since ProcessSession doesn't do so for empty flow files
+      logger_->log_debug("ReadCallback workaround on empty flow file, because DropEmptyFlowFiles is false, uuid: %s", flowFile->getUUIDStr());
+      callback.process(nullptr);
+    }
+    if (!callback.called_ && dropEmptyFlowFiles) {
+      logger_->log_info("Deprecated behavior, use connections to drop empty flow files! Dropped empty flow file with uuid: %s", flowFile->getUUIDStr());
```

Comment:
I like this log message, but I would prefer moving this to connection to avoid duplicating to multiple processors. 
Please also change the log level to debug (it's for developers, not operations)

Reply:
We can not move this to Connection, since the data loss happens in the processors, and no Connections have any way of knowing whether the empty flow file was processed or not.

Reply:
Fair point :)